docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,794 @@
+          entity_type VARCHAR(255) NOT NULL,
+          vendor VARCHAR(255) NOT NULL,
+          management_interface_id ManagementInterfaceID[] DEFAULT ARRAY['O2IMS']::ManagementInterfaceID[],
+          pk_notification_field TEXT[] DEFAULT ARRAY['resource_type_id']::TEXT[],
+          created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
+          CONSTRAINT fk_alarm_dictionary_version FOREIGN KEY (alarm_dictionary_version) REFERENCES versions(version_number)
+);
+
+-- Table: alarm_definitions
+DROP TABLE IF EXISTS alarm_definitions CASCADE;
+CREATE TABLE alarm_definitions (
+           alarm_definition_id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+           alarm_name VARCHAR(255) NOT NULL,
+           alarm_last_change VARCHAR(50) NOT NULL,
+           alarm_description TEXT NOT NULL,
+           proposed_repair_actions TEXT NOT NULL,
+           alarm_dictionary_version VARCHAR(50) NOT NULL, -- Links alarm_dictionary and alarm_definitions
+           alarm_additional_fields JSONB,
+           alarm_change_type AlarmLastChangeType DEFAULT 'added' NOT NULL,
+           clearing_type ClearingType DEFAULT 'automatic' NOT NULL,
+           management_interface_id ManagementInterfaceID[] DEFAULT ARRAY['O2IMS']::ManagementInterfaceID[],
+           pk_notification_field TEXT[] DEFAULT ARRAY['alarm_definition_id']::TEXT[],
+           created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
+           CONSTRAINT fk_alarm_definitions_version FOREIGN KEY (alarm_dictionary_version) REFERENCES versions(version_number),
+           CONSTRAINT unique_alarm_name_last_change UNIQUE (alarm_name, alarm_last_change)
+);
+
+-- Table: probable_causes
+DROP TABLE IF EXISTS probable_causes CASCADE;
+CREATE TABLE probable_causes (
+             probable_cause_id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+             alarm_definition_id UUID UNIQUE,
+             FOREIGN KEY (alarm_definition_id) REFERENCES alarm_definitions(alarm_definition_id) ON DELETE CASCADE
+);
+
+-- Create a new entry for probable cause for each new alarm_definition_id
+CREATE OR REPLACE FUNCTION insert_probable_cause()
+    RETURNS TRIGGER AS $$
+BEGIN
+    -- Insert a new row into probable_causes with the alarm_definition_id from the new alarm_definition
+    INSERT INTO probable_causes (alarm_definition_id)
+    VALUES (NEW.alarm_definition_id);
+
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER trg_insert_probable_cause
+    AFTER INSERT ON alarm_definitions
+    FOR EACH ROW
+EXECUTE FUNCTION insert_probable_cause();
+
+
+-- Table: alarm_subscription_info
+DROP TABLE IF EXISTS alarm_subscription_info CASCADE;
+CREATE TABLE alarm_subscription_info (
+             subscription_id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+             consumer_subscription_id UUID,
+             filter TEXT,
+             callback TEXT NOT NULL,
+             largest_number_alarm_event_seen_so_far BIGINT NOT NULL DEFAULT 0,
```

Comment:
`largest_number_alarm_event_seen_so_far` is a bit wordy.  Perhaps `event_cursor`?

Reply:
Yep too wordy! Updated to `event_cursor`. (will add the comments for each col during dev to explain them better)